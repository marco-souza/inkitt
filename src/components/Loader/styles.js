import { StyleSheet } from "react-native"
/**
 * Styles used on the application
 */
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center"
    }
})

export default styles
