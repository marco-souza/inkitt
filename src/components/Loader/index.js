import React from "react"
import { Pulse } from "react-native-loader"
import { View, Text } from "react-native"
import styles from "./styles"

type Props = {
    error: ?string
}

class Component extends React.Component<Props> {
    render() {
        const errorMessage: string = this.props.error || ""
        return (
            <View
                style={styles.container}
            >
                <Text> Loading... </Text>
                <Pulse size={20} color="#bbb" />
                <Text> {errorMessage} </Text>
            </View>
        )
    }
}

export default Component
