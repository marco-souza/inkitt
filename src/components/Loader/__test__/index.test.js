import React from "react"
import { shallow, render } from "enzyme"
import Loader from "../index"

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer"

describe("Test Loader Component", () => {
    it("Test Loader with error", () => {
        const errorMessage = "Test",
            tree = renderer.create(<Loader error={errorMessage} />).toJSON()

        expect(tree).toMatchSnapshot()

        // expect(shallow(Loader).contains(innerContent)).toBe(true)
    })

    it("Test without error message", () => {
        const tree = renderer.create(<Loader />).toJSON()

        expect(tree).toMatchSnapshot()

        // expect(shallow(Loader).contains(snp)).toBe(true)
    })
})
