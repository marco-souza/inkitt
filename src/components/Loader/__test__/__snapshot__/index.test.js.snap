exports[`Test Loader with error`] = `
<View
  style={
    Object {
        flex: 1,
        justifyContent: "center",
        alignItems: "center"
    }
  }>

  <Text>
    Loading...
  </Text>

  <Pulse size={20} color="#bbb" />

  <Text>
    Test
  </Text>

</View>
`;

exports[`Test Loader with error`] = `
<View
  style={
    Object {
        flex: 1,
        justifyContent: "center",
        alignItems: "center"
    }
  }>

  <Text>
    Loading...
  </Text>

  <Pulse size={20} color="#bbb" />

  <Text></Text>

</View>
`;
